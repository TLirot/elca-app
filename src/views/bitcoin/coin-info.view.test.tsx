import { CoinInfoView, CoinInfoViewProps } from "./coin-info.view";
import { shallow, ShallowWrapper } from "enzyme";

const defaultProps: CoinInfoViewProps = {
    code: "GBP",
    rate: "123",
    description: "GBP Description",
    updateISOTime: "2021-08-07T12:45:00+00:00",
    rate_float: 123,
}

const initShallowComponent = (customProps: Partial<CoinInfoViewProps> = {}):
    ShallowWrapper<CoinInfoViewProps, Record<never, never>, CoinInfoView> => {
    const props = {
        ...defaultProps,
        ...customProps,
    };

    return shallow(<CoinInfoView {...props} />)
}

describe("CoinInfoView =>", () => {

    beforeEach(() => {
        jest.restoreAllMocks();
    });

    it("Should render correctly", () => {
        const wrapper = initShallowComponent();

        expect(wrapper).toMatchSnapshot();
    })
});
