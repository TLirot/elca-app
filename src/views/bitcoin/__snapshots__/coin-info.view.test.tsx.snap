// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CoinInfoView => Should render correctly 1`] = `
<Card
  title="GBP / GBP Description"
>
  <Row
    className="justify-content-between"
  >
    <Col
      className="col-auto"
    >
      <img
        alt="bitcoin"
        src="https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcT2WR24fnzSsiHf1TmpIWQn_E3qgJTLBcsK5w&usqp=CAU"
        style={
          Object {
            "height": "50px",
            "width": "250px",
          }
        }
      />
    </Col>
    <Col
      className="col-auto align-self-center"
    >
      123 â‚¬
    </Col>
  </Row>
  <Row
    className="justify-content-end"
  >
    <CoinDiffView
      rate_float={123}
    />
  </Row>
  <hr />
  <Row>
    <Col
      className="col-auto"
    >
      Aug-07 14:45
    </Col>
  </Row>
</Card>
`;
